homeassistant/components/mqtt/binary_sensor.py

```diff
@@ -123,17 +125,19 @@ class MqttBinarySensor(MqttEntity, BinarySensorEntity, RestoreEntity):
     """Representation a binary sensor that is updated by MQTT."""
 
     _entity_id_format = binary_sensor.ENTITY_ID_FORMAT
-
-    def __init__(self, hass, config, config_entry, discovery_data):
+    _expired: bool | None
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: ConfigType,
+        config_entry: ConfigEntry,
+        discovery_data: DiscoveryInfoType | None,
+    ) -> None:
         """Initialize the MQTT binary sensor."""
         self._state: bool | None = None
-        self._expiration_trigger = None
-        self._delay_listener = None
-        expire_after = config.get(CONF_EXPIRE_AFTER)
-        if expire_after is not None and expire_after > 0:
-            self._expired = True
-        else:
-            self._expired = None
+        self._expiration_trigger: Callable[..., None] | None = None
+        self._delay_listener: Callable[..., None] | None = None
```

Comment:
`...` is used when the arguments are unknown. In this case we know there are no arguments, so it's better to use `CALLBACK_TYPE`

```suggestion
        self._expiration_trigger: CALLBACK_TYPE | None = None
        self._delay_listener: CALLBACK_TYPE | None = None
```

Note: you need to import it from `homeassistant.core`